pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1172,8 +1178,11 @@ func TestReconcileWithTimeout(t *testing.T) {
 		t.Fatalf("Expected client to have at least two action implementation but it has %d", len(actions))
 	}
 
+	// The first 2 actions are looking up the task and pipeline. The TaskRun is the 3rd action.
+	t.Log("actions", clients.Pipeline.Actions())
+
 	// Check that the expected TaskRun was created
-	actual := actions[1].(ktesting.CreateAction).GetObject().(*v1beta1.TaskRun)
+	actual := actions[2].(ktesting.CreateAction).GetObject().(*v1beta1.TaskRun)
```

Comment:
Is there some way we could loop and find the right action, instead of pulling out the one we expect to be right by index? This feels really brittle to implementation changes.